samples/fabric/fabric-cli/src/commands/asset/exchange/is-locked.ts

```diff
@@ -0,0 +1,179 @@
+        return
+    }
+    if (!options['contract-id'] && (!options['param'] || options['param'].split(':').length!=2 || !options['recipient'])) {
+      print.error(`Required either --contract-id or both recipient userid in --recipient, and asset type:id in --param for non-fungible assets`)
+      return
+    }
+    
+    let contractId: string = null, params
+    if (options['contract-id']) {
+        contractId = options['contract-id']
+    }
+    else {
+      params = options['param'].split(':')
+    }
+
+    const netConfig = getNetworkConfig(options['target-network'])
+    if (!netConfig.connProfilePath || !netConfig.channelName || !netConfig.chaincode) {
+      print.error(
+        `Please use a valid --target-network. No valid environment found for ${options['target-network']} `
+      )
+      return
+    }
+
+    const network = await fabricHelper({
+      channel: netConfig.channelName,
+      contractName: netConfig.chaincode,
+      connProfilePath: netConfig.connProfilePath,
+      networkName: options['target-network'],
+      mspId: netConfig.mspId,
+      userString: options['locker']
+    })
+    
+    var funcToCall = AssetManager.isAssetLockedInHTLCqueryUsingContractId
+    var asset = 'Asset'
+    
+    if (options['fungible']) {
+        funcToCall = AssetManager.isFungibleAssetLockedInHTLC
+        asset = 'Fungible Asset'
+    }
+
+    const spinner = print.spin(`Asset Exchange: Query ${asset} Is Locked:\n`)
+
+    if (options['contract-id']) {
+        try {
+            spinner.info(`Querying ${asset} Is Locked with ContractId: ${contractId}`)
+            const res = await funcToCall(network.contract,
+                          contractId)
+            if (!res) {
+                throw new Error()
+            }
+            spinner.info(`${asset} Is Locked Response: ${res}`)
+            spinner.succeed('Asset Exchange: Query Complete.')
+        } catch(error) {
+            print.error(`Could not query ${asset} is-locked in ${options['target-network']}`)
+            spinner.fail(`Error`)
+        }
+    } else {
+        try {
+            const lockerId = await network.wallet.get(options['locker'])
+            const lockerCert = Buffer.from((lockerId).credentials.certificate).toString('base64')
+            const recipientId = await network.wallet.get(options['recipient'])
+            const recipientCert = Buffer.from((recipientId).credentials.certificate).toString('base64')
+            spinner.info(`Querying ${asset} Is Locked with params: ${params[0]}, ${params[1]} locked by ${options.locker} for ${options.recipient}`)
+            const res = await AssetManager.isAssetLockedInHTLC(network.contract,
+                          params[0],
+                          params[1],
+                          recipientCert,
+                          lockerCert)
+            if (!res) {
+                throw new Error()
+            }
+            spinner.info(`${asset} Is Locked Response: ${res}`)
```

Comment:
Print lock hash here too? Is that available from the API?

Reply:
It's not directly available from IsLock API, a separate query needs to be called to fetch the hash.